### ISSUE TITLE
bug:  groupby result not merged in cluster mode

### ISSUE THREAD

**Issue by sandflee** [datafuselabs/databend#8290]
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

xx

### What's Wrong?

run a sql in 3 nodes cluster, generates result like
```
20221010, 10
20221010, 20
20221010, 30
```
while the right reulst is `20221010, 60`, seems we distribute data by `p_date` and `author_id`, and should add a final merge in root fragment

sql
```
SELECT a.d_f_date,
       sum(a.m_live_pay_unrisk_order_amt) AS sqlrewriter0
FROM
  (SELECT `p_date` AS `d_f_date`,
          author_id,
          coalesce(SUM(`pay_unrisk_order_amt`), 0) AS `m_live_pay_unrisk_order_amt`
   FROM tableA
   WHERE p_date BETWEEN '20211001' AND '20211115'
     AND pay_unrisk_order_amt < 2000000
     AND pay_unrisk_order_amt > 0
   GROUP BY `p_date`,
            author_id) AS a
GROUP BY a.d_f_date
ORDER BY a.d_f_date,
         sqlrewriter0
LIMIT 100
;
```

```
Fragment 0:
  DataExchange: Shuffle
  Exchange Sink: fragment id: [1]
    Aggregate(Partial): group items: [199, 0], aggregate functions: [SUM(179)]
      Filter: [>=($1, 20211001), <=($1, 20211115), <($2, 2000000), >($2, 0)]
        TableScan: [tableA]

Fragment 1:
  DataExchange: Merge
  Exchange Sink: fragment id: [2]
    EvalScalar: [$0]
      Aggregate(Final): group items: [199], aggregate functions: [sum(200)]
        Aggregate(Partial): group items: [199], aggregate functions: [sum(200)]
          EvalScalar: [multi_if(is_not_null($0), assume_not_null($0), true, 0, NULL)]
            Aggregate(Final): group items: [199, 0], aggregate functions: [SUM(179)]
              Exchange Source: fragment id: [0]

Fragment 2:
  Limit: [100], Offset: [0]
    Sort: [199 ASC, 204 ASC], Limit: [100]
      Exchange Source: fragment id: [1]
```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by sundy-li** [comment]
Related to distributed pipeline builder. cc @leiysky

**Comment by sandflee** [comment]
@leiysky , could you explain how u fixed #8290 and #8292 ?
